src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -89,59 +86,18 @@ an out-of-memory error if allocation fails.
   .fn(async t => {
     const { oom, usage, size } = t.params;
 
-    const buffer = t.expectGPUError(
-      'out-of-memory',
-      () => t.device.createBuffer({ mappedAtCreation: true, size, usage }),
-      oom
-    );
-
-    const f = () => buffer.getMappedRange();
+    const f = () => t.device.createBuffer({ mappedAtCreation: true, size, usage });
 
-    let mapping: ArrayBuffer | undefined = undefined;
     if (oom) {
       // getMappedRange is normally valid on OOM buffers, but this one fails because the
       // (default) range is too large to create the returned ArrayBuffer.
       t.shouldThrow('RangeError', f);
     } else {
-      mapping = f();
-    }
-
-    // Should be valid because buffer is mapped, regardless of OOM.
-    buffer.unmap();
-    if (mapping !== undefined) {
-      t.expect(mapping.byteLength === 0, 'Mapping should be detached');
+      const buffer = f();
+      const mapping = buffer.getMappedRange();
+      buffer.unmap();
+      if (mapping !== undefined) {
+        t.expect(mapping.byteLength === 0, 'Mapping should be detached');
+      }
```

Comment:
getMappedRange isn't allowed to return `undefined`.
```suggestion
      t.expect(mapping.byteLength === size, 'Mapping should be successful');
      buffer.unmap();
      t.expect(mapping.byteLength === 0, 'Mapping should be detached');
```